syntax = "proto3";

package auth;

import "google/protobuf/empty.proto";

service AuthService {
  rpc Register (RegisterRequest) returns (google.protobuf.Empty);
  rpc Login (LoginRequest) returns (AuthResponse);
  rpc Validate (ValidateRequest) returns (ValidateResponse);
}

message RegisterRequest {
  string username = 1;
  string password = 2;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message AuthResponse {
  string token = 1;
  int64  expires_at = 2;
}

message ValidateRequest {
  string token = 1;
}

message ValidateResponse {
  bool valid  = 1;
  string user = 2;
}
